From 2fb6cc4a116f62b9a64ab2595826335381edd04e Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Mon, 11 Jun 2018 15:54:03 +0800
Subject: [PATCH 3957/5242] MLK-18574: ASoC: fsl_spdif: support 192kHz for rx
 in imx8

commit  6f3eb277b2f2c8ade0ce7c310fa7fac2565a9d04 from
https://source.codeaurora.org/external/imx/linux-imx.git

The ipg clock is higher enough to support 192kHz in imx8

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_spdif.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sound/soc/fsl/fsl_spdif.c b/sound/soc/fsl/fsl_spdif.c
index a84b77d..8e9e142 100644
--- a/sound/soc/fsl/fsl_spdif.c
+++ b/sound/soc/fsl/fsl_spdif.c
@@ -53,6 +53,7 @@ struct fsl_spdif_soc_data {
 	u32 rx_burst;
 	u32 interrupts;
 	u64 tx_formats;
+	u64 rx_rates;
 };
 
 /*
@@ -132,6 +133,7 @@ struct fsl_spdif_priv {
 	.rx_burst = FSL_SPDIF_RXFIFO_WML,
 	.interrupts = 1,
 	.tx_formats = FSL_SPDIF_FORMATS_PLAYBACK,
+	.rx_rates = FSL_SPDIF_RATES_CAPTURE,
 	.constrain_period_size = false,
 };
 
@@ -142,6 +144,7 @@ struct fsl_spdif_priv {
 	.rx_burst = FSL_SPDIF_RXFIFO_WML,
 	.interrupts = 1,
 	.tx_formats = FSL_SPDIF_FORMATS_PLAYBACK,
+	.rx_rates = FSL_SPDIF_RATES_CAPTURE,
 	.constrain_period_size = false,
 };
 
@@ -157,6 +160,7 @@ struct fsl_spdif_priv {
 	.rx_burst = 2,
 	.interrupts = 2,
 	.tx_formats = SNDRV_PCM_FMTBIT_S24_LE,
+	.rx_rates = FSL_SPDIF_RATES_CAPTURE,
 	.constrain_period_size = true,
 };
 
@@ -167,6 +171,7 @@ struct fsl_spdif_priv {
 	.rx_burst = 2,
 	.interrupts = 2,
 	.tx_formats = SNDRV_PCM_FMTBIT_S24_LE,
+	.rx_rates = (FSL_SPDIF_RATES_CAPTURE | SNDRV_PCM_RATE_192000),
 	.constrain_period_size = true,
 };
 
@@ -1333,6 +1338,8 @@ static int fsl_spdif_probe(struct platform_device *pdev)
 	spdif_priv->cpu_dai_drv.name = dev_name(&pdev->dev);
 	spdif_priv->cpu_dai_drv.playback.formats =
 				spdif_priv->soc->tx_formats;
+	spdif_priv->cpu_dai_drv.capture.rates =
+				spdif_priv->soc->rx_rates;
 
 	/* Get the addresses and IRQ */
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-- 
1.7.9.5

